File: socketServer.py
# -*- coding:utf-8 -*-
"""
Dec: server端，可以用Go语言的client接收，也可以用Python语言的client接收
Created on: 2018.10.11
Author: Iflier
"""

import time
import socket
import argparse
from datetime import datetime


ap = argparse.ArgumentParser()
ap.add_argument("-p", "--port", type=int, default=60000, help="Specify server open which one port")
args = vars(ap.parse_args())

ADDRESS = ("127.0.0.1", args["port"])

sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
sock.bind(ADDRESS)
sock.listen()
conn, addr = sock.accept()
print("addr: {0}".format(addr))

for _ in range(10):
    numBytesWritten = conn.send("Yes.".encode())
    print("On {0}, written {1:^5,d} bytes.".format(datetime.now().strftime("%c"), numBytesWritten))
    time.sleep(1.0)
    receivedBytes = conn.recv(128)
    print("On {0}, received: {1}".format(datetime.now().strftime("%c"), receivedBytes.decode()))

numBytesWritten = conn.send("exit".encode())
print("On {0}, written {1:^5,d} bytes.".format(datetime.now().strftime("%c"), numBytesWritten))
conn.close()
sock.close()
print("Done.")

File: socketServer.go
package main

import (
	"flag"
	"fmt"
	"log"
	"net"
	"time"
)

func main() {
	var sendTimes = 0

	var port = flag.String("port", "60000", "Specify open port.")
	flag.Parse()

	var ADDRESS = net.JoinHostPort("127.0.0.1", *port)
	receiveBuf := make([]byte, 10, 16)
	listener, err := net.Listen("tcp", ADDRESS)
	dealErr(err)
	fmt.Printf("Listener network address: %v\n", listener.Addr().String())
	conn, err := listener.Accept()
	dealErr(err)
	fmt.Printf("Remote network address: %v\n", conn.RemoteAddr().String())
	// 从client端发送字节，然后读取client返回的字节，就这样循环10次
	for sendTimes < 10 {
		numBytesWritten, err := conn.Write([]byte("Yes."))
		dealErr(err)
		fmt.Printf("On %v, written %v bytes.\n", time.Now().String(), numBytesWritten)
		time.Sleep(1 * time.Second)
		numBytesRead, err := conn.Read(receiveBuf)
		dealErr(err)
		fmt.Printf("On %v, received %v bytes: %v\n", time.Now().String(), numBytesRead, string(receiveBuf[:numBytesRead]))
		sendTimes++
	}
	numBytesWritten, err := conn.Write([]byte("exit"))
	dealErr(err)
	fmt.Printf("On %v, written %v bytes.\n", time.Now().String(), numBytesWritten)
	conn.Close()
	fmt.Println("Done.")
}

func dealErr(err error) {
	if err != nil {
		log.Fatal(err.Error())
	}
}
